src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -48,6 +49,26 @@ protected ServiceHubServiceBase(Stream stream, IServiceProvider serviceProvider)
             Rpc.Disconnected += OnRpcDisconnected;
         }
 
+        protected ServiceHubServiceBase(IServiceProvider serviceProvider, Stream stream)
+        {
+            _instanceId = Interlocked.Add(ref s_instanceId, 1);
+
+            // in unit test, service provider will return asset storage, otherwise, use the default one
+            AssetStorage = (AssetStorage)serviceProvider.GetService(typeof(AssetStorage)) ?? AssetStorage.Default;
+
+            Logger = (TraceSource)serviceProvider.GetService(typeof(TraceSource));
+            Logger.TraceInformation($"{DebugInstanceString} Service instance created");
+
+            _cancellationTokenSource = new CancellationTokenSource();
+            CancellationToken = _cancellationTokenSource.Token;
+
+            // due to this issue - https://github.com/dotnet/roslyn/issues/16900#issuecomment-277378950
+            // all sub type must explicitly start JsonRpc once everything is
+            // setup
+            Rpc = new JsonRpc(stream, stream, this);
+            Rpc.Disconnected += OnRpcDisconnected;
```

Comment:
this is missing -  Rpc.JsonSerializer.Converters.Add(AggregateJsonConverter.Instance);